application/src/main/java/org/togetherjava/tjbot/commands/free/FreeCommand.java

```diff
@@ -0,0 +1,330 @@
+            }
+            channelMonitor.setChannelBusy(guildEvent.getChannel().getIdLong(),
+                    guildEvent.getAuthor().getIdLong());
+            displayStatus(channelMonitor.getStatusChannelFor(guildEvent.getGuild()));
+            guildEvent.getMessage().reply(UserStrings.NEW_QUESTION.message()).queue();
+        }
+    }
+
+    private Optional<Message> getStatusMessageIn(@NotNull TextChannel channel) {
+        if (channelToStatusMessage.containsKey(channel.getIdLong())) {
+            Long id = channelToStatusMessage.get(channel.getIdLong());
+            if (id == null) {
+                return Optional.empty();
+            }
+            Message message = channel.getHistoryAround(id, 1).complete().getMessageById(id);
+            if (message == null) {
+                return Optional.empty();
+            }
+            return Optional.of(message);
+        }
+        return findExistingStatusMessage(channel);
+    }
+
+    private Optional<Message> findExistingStatusMessage(@NotNull TextChannel channel) {
+        // will only run when bots starts, afterwards its stored in a map
+        Optional<Message> result = channel.getHistory()
+            .retrievePast(100)
+            .map(history -> history.stream()
+                .filter(message -> !message.getEmbeds().isEmpty())
+                .filter(message -> message.getAuthor().equals(channel.getJDA().getSelfUser()))
+                // .filter(message -> STATUS_TITLE.equals(message.getEmbeds().get(0).getTitle()))
+                .findFirst())
+            .complete();
+
+        if (result.isPresent()) {
+            channelToStatusMessage.put(channel.getIdLong(), result.get().getIdLong());
+        } else {
+            channelToStatusMessage.put(channel.getIdLong(), null);
+        }
+        return result;
+    }
+
+    private void initChannelsToMonitor() {
+        Config.getInstance()
+            .getFreeCommandConfig()
+            .stream()
+            .map(FreeCommandConfig::getMonitoredChannels)
+            .flatMap(Collection::stream)
+            .forEach(channelMonitor::addChannelToMonitor);
+    }
+
+    private void initStatusMessageChannels(@NotNull final JDA jda) {
+        Config.getInstance()
+            .getFreeCommandConfig()
+            .stream()
+            .map(FreeCommandConfig::getStatusChannel)
+            .distinct() // not necessary? (validates user input, since input is from file)
+            .map(jda::getTextChannelById)
+            .filter(Objects::nonNull) // not necessary? this will hide errors in the config file
+            .forEach(channelMonitor::addChannelForStatus);
+    }
+
+    private void initStatusMessages(@NotNull final JDA jda) {
+        // not currently working, attempts to find the existing status message. (for all guilds)
+        channelMonitor.statusIds()
+            .map(jda::getTextChannelById)
+            .filter(Objects::nonNull) // not necessary? this will hide errors in the config file
```

Comment:
I'm not sure if it should be removed or not.
It will hide errors in the config.json file.

Do we want notifications of errors?
Do we want the system to always work 'as best as it can' or crash with warning and wait for the mistake to get corrected

Reply:
as of now, the code pretty much crashes on purpose if the config is not exactly as expected. this is a bit annoying during development but very helpful at release.

i would say, stick to the existing pattern.

that said, if the config is correct but the server is not configured correctly (for example there is no "Muted" role although its listed in the config), then we used `warn` or `error` logs so far.

Reply:
fixed